app/src/data/host-context.tsx

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react'
+import { QuestionType } from '../models/QuestionType'
+import { ResponseValue } from '../models/ResponseValue'
+import { ApiWrapper } from './ApiWrapper'
+
+const HostDashboardContext = React.createContext({})
+const hostsFetcher = new ApiWrapper()
+
+interface HostDashboardData {
+    hostQuestions: Array<QuestionType>
+    hostResponses: Array<ResponseValue>
+    loaderState: {
+        loading: boolean
+        message: string
+    }
+}
+
+enum HostDashboardActionType {
+    BeginFetchQuestions,
+    FinishFetchQuestions,
+    GetHostById,
+    isLoading,
+    BeginPostAnswer,
+    FinishPostAnswer,
+    Error,
+}
+
+interface HostDashboardAction {
+    type: HostDashboardActionType
+    payload?: HostDashboardData | Array<QuestionType> | boolean | string
+}
+
+const initialState: HostDashboardData = {
+    hostQuestions: [],
+    hostResponses: [],
+    loaderState: {
+        loading: false,
+        message: '',
+    },
+}
+
+function hostDashboardReducer(
+    state: HostDashboardData,
+    action: HostDashboardAction
+): HostDashboardData {
+    switch (action.type) {
+        case HostDashboardActionType.BeginFetchQuestions: {
+            return {
+                ...state,
```

Comment:
Since this case is for a well-defined action type, we can simplify by omitting the `payload` property from the call. In other words, `action.type === BeginFetchQuestions` already provides all the necessary information to determine what message should be displayed in the Loader. Unless you foresee that this text could vary for different dispatch calls for this action, we can alleviate the caller from needing to pass this string every time.